### PR TITLE
refactor(generator/rust): factor out some common execute code

### PR DESCRIPTION
**Pull request by codyoss**
Had to create a `NoBody` type to pass along to satisfy generics. This type is not exported. 

Fixes: #139

### CODE REVIEW

**Comment by codyoss** [comment]
@coryan The spell check is a false positive. Not sure how to get around that...

**Comment by coryan** [comment]
This worked for me:

```
[type.mustache]
extend-glob = ["*.mustache"]

[type.mustache.extend-words]
ser = "ser"
```

File: generator/templates/rust/crate/src/lib.rs.mustache [hunk 1]
```diff
@@ -106,6 +106,10 @@ impl Client {
     }
     {{/Services}}
 }
+
+#[derive(serde::Serialize)]
+#[allow(dead_code)]
+struct NoBody {}
```
Comment (coryan) [inline generator/templates/rust/crate/src/lib.rs.mustache:112]: If I was clever I would suggest a way to do this with the unit type...

File: generator/templates/rust/crate/src/lib.rs.mustache [hunk 2]
```diff
@@ -137,23 +141,36 @@ impl {{NameToPascal}} {
...
-            .map_err(Error::io)?;
-        if !res.status().is_success() {
-            let status = res.status().as_u16();
-            let headers = gax::error::convert_headers(res.headers());
-            let body = res.bytes().await.map_err(Error::io)?;
+        self.execute(builder, {{#HasBody}}Some(req{{BodyAccessor}}){{/HasBody}}{{^HasBody}}None::<NoBody>{{/HasBody}}).await
+    }
+    {{/Methods}}
+
+    async fn execute<I: serde::ser::Serialize, O: serde::de::DeserializeOwned>(
```
Comment (coryan) [inline generator/templates/rust/crate/src/lib.rs.mustache:148]: For a future PR: how do we deal with services that have an `execute()` RPC?

**Review by coryan — APPROVED** [review]

File: generator/templates/rust/crate/src/lib.rs.mustache (hunk 1 above)
Comment (codyoss) [reply generator/templates/rust/crate/src/lib.rs.mustache:112]: Yeah, I could not figure something like that out...

File: generator/templates/rust/crate/src/lib.rs.mustache (hunk 2 above)
Comment (codyoss) [reply generator/templates/rust/crate/src/lib.rs.mustache:148]: ack, cross that when we get there. I am sure we can mangle or throw an underscore in front or something.

**Pull request merged**